macros/src/actix.rs

```diff
@@ -1069,6 +1061,147 @@ pub fn emit_v2_security(input: TokenStream) -> TokenStream {
+                            if let Lit::Str(attr_value) = attr_value {
+                                if parameter_attrs
+                                    .insert(attr_name.clone(), attr_value.value())
+                                    .is_some()
+                                {
+                                    emit_warning!(
+                                        span,
+                                        "Attribute {} defined multiple times.",
+                                        attr_name
+                                    );
+                                }
+                            } else {
+                                emit_warning!(
+                                    span,
+                                    "Invalid value for named attribute: {}",
+                                    attr_name
+                                );
+                            }
+                        } else {
+                            emit_warning!(span, invalid_attr_msg);
+                        }
+                    } else {
+                        emit_error!(span, invalid_attr_msg);
+                    }
+                }
+                _ => {
+                    emit_error!(span, invalid_attr_msg);
+                }
+            }
+        }
+    }
+
+    let quoted_description = quote_option(parameter_attrs.get("description"));
+    let name_string = name.to_string();
+    let quoted_name = parameter_attrs.get("name").unwrap_or(&name_string);
+
+    let quoted_format = if let Some(format) = parameter_attrs.get("format") {
+        match &*format.clone() {
+            "int32" => quote! { Some(paperclip::v2::models::DataTypeFormat::Int32) },
+            "int64" => quote! { Some(paperclip::v2::models::DataTypeFormat::Int64) },
+            "float" => quote! { Some(paperclip::v2::models::DataTypeFormat::Float) },
+            "double" => quote! { Some(paperclip::v2::models::DataTypeFormat::Double) },
+            "byte" => quote! { Some(paperclip::v2::models::DataTypeFormat::Byte) },
+            "binary" => quote! { Some(paperclip::v2::models::DataTypeFormat::Binary) },
+            "data" => quote! { Some(paperclip::v2::models::DataTypeFormat::Date) },
+            "datetime" => quote! { Some(paperclip::v2::models::DataTypeFormat::DateTime) },
+            "password" => quote! { Some(paperclip::v2::models::DataTypeFormat::Password) },
+            "url" => quote! { Some(paperclip::v2::models::DataTypeFormat::Url) },
+            "uuid" => quote! { Some(paperclip::v2::models::DataTypeFormat::Uuid) },
+            "ip" => quote! { Some(paperclip::v2::models::DataTypeFormat::Ip) },
+            "ipv4" => quote! { Some(paperclip::v2::models::DataTypeFormat::IpV4) },
+            "ipv6" => quote! { Some(paperclip::v2::models::DataTypeFormat::IpV6) },
+            "other" => quote! { Some(paperclip::v2::models::DataTypeFormat::Other) },
```

Comment:
Could we add a `TryFrom<&str>` to `DataTypeFormat` ?

Reply:
I might be wrong but if we use a `FromStr` implementation would it still fail at compile time ?

Another solution would be to use [FromMeta](https://docs.rs/darling/latest/darling/trait.FromMeta.html) but it means adding darling as a dependency.

Reply:
We would be better !